kubernetes/kubernetes_service.go

```diff
@@ -165,33 +156,27 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 	}
 	serviceDetails.Pods = podsResponse.pods
 
+	// Finally, after we get the pods we can start fetching deployments, then narrow down the list
+	deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(emptyListOptions)
```

Comment:
Why can't we do this fetch in parallel? It is a fetch without any list option, so I guess I can be placed as it was before.
I guess is the Line 164 the one that have to wait for pods and deployments.

Reply:
Oh, you're right. I iterated several times on this code change and didn't notice that it could be kept parallelized.

Reply:
Done, it's still in the main thread (no need to create channel here that would make unnecessary resources be used) but now is performed before waiting for other channels return so it's parallel.